aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/AWSS3StorageService.java

```diff
@@ -147,6 +150,28 @@ public TransferObserver uploadFile(
         return transferUtility.upload(bucket, serviceKey, file, metadata);
     }
 
+    /**
+     * Begin uploading an inputStream.
+     * @param serviceKey S3 service key
+     * @param inputStream Target InputStream
+     * @param metadata Object metadata to associate with upload
+     * @return A transfer observer
+     * @throws IOException An IOException thrown during the process writing an InputStream into a file
+     */
+    @NonNull
+    public TransferObserver uploadInputStream(
+            @NonNull String serviceKey,
+            @NonNull InputStream inputStream,
+            @NonNull ObjectMetadata metadata
+    ) throws IOException {
+        startServiceIfNotAlreadyStarted();
+        UploadOptions uploadOptions = UploadOptions.builder()
```

Comment:
We do have `UploadOptions` class within Storage SDK. This class serves as a builder to pass parameters to `upload()` method.

Reply:
Ahh, right, you added `UploadOptions` in the 2.19.2 SDK release.  THe build is failing right now though is because Amplify still points to 2.19.1, which doesn't have `UploadOptions`.  Can you update to 2.19.2 [here](https://github.com/aws-amplify/amplify-android/blob/3ab05442608fb26508be233b88bc68fef0014513/build.gradle#L54)?  Once you get the build passing, I think this PR is ready to ship!